ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoTransactionBuilder.kt

```diff
@@ -0,0 +1,228 @@
+     */
+    fun setTimeWindowBetween(from: Instant, until: Instant): UtxoTransactionBuilder
+
+    /**
+     * Sets the transaction time window to be valid at the specified [Instant] with the specified tolerance.
+     *
+     * @param midpoint The [Instant] representing the midpoint of the time window.
+     * @param tolerance The [Duration] representing the tolerance of the time window, which is applied before and after the midpoint.
+     * @return Returns a [UtxoTransactionBuilder] including the specified time window.
+     */
+    fun setTimeWindowBetween(midpoint: Instant, tolerance: Duration): UtxoTransactionBuilder
+
+    /**
+     * Signs the transaction with any required signatories that belong to the current node.
+     *
+     * @return Returns a [UtxoSignedTransaction] with signatures for any required signatories that belong to the current node.
+     */
+    fun sign(): UtxoSignedTransaction
+
+    /**
+     * Signs the transaction with the specified signatory keys.
+     *
+     * @param signatories The signatories expected to sign the current transaction.
+     * @return Returns a [UtxoSignedTransaction] with signatures for the specified signatory keys.
+     */
+    fun sign(signatories: Iterable<PublicKey>): UtxoSignedTransaction
+
+    /**
+     * Signs the transaction with the specified signatory keys.
+     *
+     * @param signatories The signatories expected to sign the current transaction.
+     * @return Returns a [UtxoSignedTransaction] with signatures for the specified signatory keys.
+     */
+    fun sign(vararg signatories: PublicKey): UtxoSignedTransaction
+
+    /**
+     * Verifies the current transaction.
+     */
+    fun verify()
+
+    /**
+     * Verifies and signs the transaction with any required signatories that belong to the current node.
+     *
+     * @return Returns a [UtxoSignedTransaction] with signatures for any required signatories that belong to the current node.
+     */
+    fun verifyAndSign(): UtxoSignedTransaction
+
+    /**
+     * Verifies and signs the transaction with the specified signatory keys.
+     *
+     * @param signatories The signatories expected to sign the current transaction.
+     * @return Returns a [UtxoSignedTransaction] with signatures for the specified signatory keys.
+     */
+    fun verifyAndSign(signatories: Iterable<PublicKey>): UtxoSignedTransaction
+
+    /**
+     * Verifies and signs the transaction with the specified signatory keys.
+     *
+     * @param signatories The signatories expected to sign the current transaction.
+     * @return Returns a [UtxoSignedTransaction] with signatures for the specified signatory keys.
+     */
+    fun verifyAndSign(vararg signatories: PublicKey): UtxoSignedTransaction
+
+    /**
+     * Builds the current transaction into a [UtxoWireTransaction].
+     *
+     * @return Returns a [UtxoWireTransaction] from the current [UtxoTransactionBuilder].
+     */
+    fun toWireTransaction(): UtxoWireTransaction
```

Comment:
Not sure that needs to be on the public API - users should only have to deal in signed and ledger tx.